tensorflow_probability/python/stats/sample_stats.py

```diff
@@ -906,30 +925,28 @@ def _prepare_window_args(x, low_indices=None, high_indices=None, axis=0):
   high_indices = high_indices + tf.zeros_like(low_indices)
   low_indices = low_indices + tf.zeros_like(high_indices)
 
-  # TODO(axch): Support batch low and high indices.  That would
-  # complicate this shape munging (though tf.gather should work
-  # fine).
-
-  # We want to place `low_counts` and `high_counts` at the `axis`
-  # position, so we reshape them to shape `[1, 1, ..., 1, N, 1, ...,
-  # 1]`, where the `N` is at `axis`.  The `counts_shp`, below,
-  # is this shape.
-  size = ps.size(high_indices)
-  counts_shp = ps.one_hot(
-      axis, depth=ps.rank(x), on_value=size, off_value=1)
-
-  low_counts = tf.reshape(tf.cast(low_indices, dtype=x.dtype),
-                          shape=counts_shp)
-  high_counts = tf.reshape(tf.cast(high_indices, dtype=x.dtype),
-                           shape=counts_shp)
-  return low_indices, high_indices, low_counts, high_counts
-
-
-def _safe_average(totals, counts):
-  # This tf.where protects `totals` from getting a gradient signal
-  # when `counts` is 0.
-  safe_totals = tf.where(~tf.equal(counts, 0), totals, 0)
-  return tf.where(~tf.equal(counts, 0), safe_totals / counts, 0)
+  indices_shape = ps.shape(low_indices)
+  if ps.rank(low_indices) < ps.rank(x):
```

Comment:
We can't write this like this. prefer_static will return Tensors if the inputs are not statically known, which cannot appear in a conditional like this. This code needs to be written without the conditional, or, as a last resort, using `tf.cond`.

Alternatively, we could require that the ranks are known, but then you need to do something like:

```
indices_rank = tf.get_static_value(ps.rank(low_indices))
if indices_rank is None:
  raise ValueError(...)
```

Reply:
I added the rank check.